### PR TITLE
DOCS-10770-X12-2.6-ESB

### PR DESCRIPTION
**Pull request by emileemooreVT**
Adding 2.6 info

### CODE REVIEW

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-batching.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
+= X12 EDI Connector 2.3 - Batch Outbound X12 Messages - Mule 4
+
+For B2B and EDI messages that are not time-sensitive, such as invoices and payment remittances, batch messages instead of transmitting transactions to your trading partners individually.
```
Comment (vijayang) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-batching.adoc]: Please rephrase as below:

For EDI messages that are not time-sensitive, such as invoice and payment remittances, you can use the Batch operation to merge multiple individual messages into a single interchange and send to your trading partners.

**Review by vijayang — CHANGES_REQUESTED** [review]

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-batching.adoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
+= X12 EDI Connector 2.3 - Batch Outbound X12 Messages - Mule 4
+
+For EDI messages that are not time-sensitive, such as invoice and payment remittances, you can use the Batch Write operation to merge multiple individual messages into a single interchange to send to your trading partners.
```
Comment (kbostrom) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-batching.adoc]: Please duplicate the edits made to the same text in edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc (applying the proper connector name, of course).
I will scan to look for anything that is different....

**Review by kbostrom — DISMISSED** [review]
ditto edits edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc

**Review by vijayang — DISMISSED** [review]

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
+= X12 EDI Connector 2.5 - Control Numbers - Mule 4
+
+Using the Write operation in the Anypoint Connector for X12 EDI Connector (X12 EDI Connector), you can generate control numbers on the generated X12 payload by:
```
Comment (valkyrie69) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc]:
```suggestion
Using the Write operation in Anypoint Connector for X12 EDI Connector (X12 EDI Connector), you can generate control numbers on the generated X12 payload by:
```

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+* Managing global control numbers through control number keys
+* Managing identifier combination control numbers
+* Using your own control number
+
+== Manage Global Control Numbers Through Control Number Keys
+
+If your organization or your trading partner have must maintain unique interchange, group, or transaction set control numbers across different identifier combinations and document types:
```
Comment (valkyrie69) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc]:
```suggestion
If your organization or your trading partner must maintain unique interchange, group, or transaction set control numbers across different identifier combinations and document types:
```

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+* Set the configuration `Use supplied values` to `false`.
+* Leverage the control number key configurations (at the configuration level or operation level) to set a key unique to the partner. +
```
Comment (valkyrie69) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc]:
```suggestion
* Leverage the control number key configurations (at the configuration level or operation level) to set a key that is unique to the partner. +
```

File: x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+image::X12-edi-control-number.jpg[X12 EDI Connector Control Numbers]
+
+== Manage Identifier Combination Control Numbers
+
+If you want the connector to automatically maintain control number sequencing based on the sender/receiver identifiers and you do not require global control numbers at a partner level, then set the Write configuration `Use supplied values` to `false`. When no value passes into the optional interchange, group, or transaction number key configurations, the sender/receiver identifier combination is the default.
+
+== Use Your Own Control Number
+
+If you must generate the control numbers and use them on the EDI payload, you can set the parameter `Use supplied values` to `true`, and map the control number on the payload in the DataWeave map.
```
Comment (valkyrie69) [inline x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc]:
```suggestion
To generate the control numbers and use them on the EDI payload, set the parameter `Use supplied values` to `true`, and map the control number on the payload in the DataWeave map.
```

File: x12-edi/2.6/modules/ROOT/pages/index.adoc [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+== Exchange Templates and Examples
+
+https://www.mulesoft.com/exchange/[Anypoint Exchange] provides templates
+that you can use as starting points for your apps and examples that illustrate a complete solution.
+
+=== X12 EDI Connector Templates in Exchange
+
+https://www.mulesoft.com/exchange/org.mule.templates//[Anypoint Exchange Templates] provide starting points for Studio projects such as:
+
+* Sets up your Outbound transactions.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/index.adoc:40]:
```suggestion
* Setting up your Outbound transactions.
```

File: x12-edi/2.6/modules/ROOT/pages/index.adoc [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+https://www.mulesoft.com/exchange/org.mule.templates/template-b2b-edi-outbound-x12-map/[Map application messages to outbound EDI X12 transactions]
+
+* Sets up your Inbound transactions.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/index.adoc:44]:
```suggestion
* Setting up your Inbound transactions.
```

File: x12-edi/2.6/modules/ROOT/pages/index.adoc [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+https://www.mulesoft.com/exchange/org.mule.templates/template-b2b-edi-inbound-x12-map/[Map inbound EDI X12 transactions to application message format]
+
+* Use this for setting up your Partner Manager content storage.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/index.adoc:48]: The lead-in sentence to this bullet point is:
[Anypoint Exchange Templates] provide starting points for Studio projects such as:

So this wording should probably be modified somehow.

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-batching.adoc [hunk 10]
```diff
@@ -0,0 +1,65 @@
+= X12 EDI Connector 2.6 - Batch Outbound X12 Messages - Mule 4
+
+For EDI messages that are not time-sensitive, such as invoice and payment remittances, use the Write batch operation to merge multiple individual messages into a single interchange to send to your trading partners.
+
+== Batching Stages
+
+Use the Anypoint Connector for X12 EDI (X12 EDI Connector) Write batch operation to merge individual X12 messages into a single transmission to send to your trading partners.
+
+You implement a batch involves two stages:
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-batching.adoc]:
```suggestion
Implementing a batch involves two stages:
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-batching.adoc [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+=== Accumulate the Transactions
+
+To accumulate transactions for a batch:
+
+. Transform the payloads into X12 messages using DataWeave and X12 as you receive application messages (such as a JSON invoice) from the backend systems.
+. Create X12 payloads using temporary control number keys in the X12 Write operation. +
++
+
+This enables sending the accumulated X12 as a batch later. The batch is created using control numbers that are not used in the final batch sent to the trading partner, for example, `<x12:write doc:name="Create a single X12 810 Invoice Txn" doc:id="767e1321-044d-4ed4-95ba-6ddc78efb0b4" config-ref="X12_EDI_Write_810" interchangeNumberKey="MYTHICAL-NTOETAIL-ISA-TMP" groupNumberKey="MYTHICAL-NTOETAIL-GS-TMP"/>`.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-batching.adoc]:
```suggestion
This enables sending the accumulated X12 as a batch at a later time. The batch is created using control numbers that are not used in the final batch that is sent to the trading partner, for example, `<x12:write doc:name="Create a single X12 810 Invoice Txn" doc:id="767e1321-044d-4ed4-95ba-6ddc78efb0b4" config-ref="X12_EDI_Write_810" interchangeNumberKey="MYTHICAL-NTOETAIL-ISA-TMP" groupNumberKey="MYTHICAL-NTOETAIL-GS-TMP"/>`.
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-config-topics.adoc [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+If you're using one or more custom schemas, locate these under
+a directory in `src/main/app` and refer to the location relative to this directory.
+For example, if an `850` schema (either overlay or full) is under `src/main/app/mypartner/850.esl`,
+the schema location is `/mypartner/850.esl`.
+
+You can configure schema locations in the Anypoint Studio XML view. The schema files are loaded when the connector is initialized, so runtime expressions such as flow variables are not supported.
+
+To configure schemas in Anypoint Studio:
+
+. Switch to the XML view by clicking Configuration XML
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-config-topics.adoc:23]:
```suggestion
. Switch to the XML view by clicking *Configuration XML*.
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+Using the Write operation in the Anypoint Connector for X12 EDI Connector (X12 EDI Connector), you can generate control numbers on the generated X12 payload by:
+
+* Managing global control numbers through control number keys
+* Managing identifier combination control numbers
+* Using your own control number
+
+== Manage Global Control Numbers Through Control Number Keys
+
+If your organization or your trading partner have must maintain unique interchange, group, or transaction set control numbers across different identifier combinations and document types:
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc:11]:
```suggestion
If your organization or your trading partner must maintain unique interchange, group, or transaction set control numbers across different identifier combinations and document types:
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+
+image::X12-edi-control-number.jpg[X12 EDI Connector Control Numbers]
+
+== Manage Identifier Combination Control Numbers
+
+If you want the connector to automatically maintain control number sequencing based on the sender/receiver identifiers and you do not require global control numbers at a partner level, then set the Write configuration `Use supplied values` to `false`. When no value passes into the optional interchange, group, or transaction number key configurations, the sender/receiver identifier combination is the default.
+
+== Use Your Own Control Number
+
+If you must generate the control numbers and use them on the EDI payload, you can set the parameter `Use supplied values` to `true`, and map the control number on the payload in the DataWeave map.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc:26]:
```suggestion
To generate the control numbers and use them on the EDI payload, set the parameter `Use supplied values` to `true`, and map the control number on the payload in the DataWeave map.
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 15]
```diff
@@ -0,0 +1,248 @@
...
+The supplied HIPAA schemas use different structures from the base X12 schemas.
+The structure used for HIPAA reflects the way that the same segments are used in different locations in a transaction set for different purposes, and often with different requirements for code sets and field usages. To help distinguish between the different uses, the HIPAA schemas use full names as segment keys. This differs from the approach used with basic X12 standards, in which the segment keys normally use a numeric prefix along with the segment tag.
+
+HIPAA schemas also include details of code sets defined for elements. Data is validated against these code sets both when parsing and writing transaction sets.
+
+== Configuring for HIPAA Usage
+
+To use X12 EDI Connector for HIPAA transaction sets you must:
+
+* Select one or more HIPAA schemas to be used.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-hipaa.adoc:33]:
```suggestion
* Select one or more HIPAA schemas to use.
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 16]
```diff
@@ -0,0 +1,248 @@
...
+
+== Modifying Schemas
+
+Due to the differences between standard X12 and HIPAA schemas the use of overlay schemas to modify a base
+definition is not supported for HIPAA. Instead, for modifications, extract the supplied HIPAA schema from inside the `x12-schemas-2.0.0.jar` file. This is found in the standard MuleSoft enterprise Maven repositories, under group ID com.mulesoft.connectors.
+You can copy a message structure schema from this JAR file (it contains both standard X12 and HIPAA schemas) and modify the extracted schema to use it directly. Unlike X12 schemas that use a base set of segment, composite, and element definitions, the HIPAA schemas are self-contained. This makes it easy to make changes to the schema without working with multiple files.
+
+== Recommended Types of Validations from WEDI
+
+There are two types of validations to use for integrity testing.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-hipaa.adoc:229]:
```suggestion
There are two types of validations to use for integrity testing:
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 17]
```diff
@@ -0,0 +1,248 @@
...
+
+* Type 1: EDI syntax integrity testing
+
+Testing of the EDI file for valid segments, segment order, element attributes, testing for numeric values in numeric data elements, validation of X12 or NCPDP syntax, and compliance with X12 and NCPDP rules.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-hipaa.adoc:233]:
```suggestion
Tests the EDI file for valid segments, segment order, element attributes, as well as testing for numeric values in numeric data elements, validation of X12 or NCPDP syntax, and compliance with X12 and NCPDP rules.
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-hipaa.adoc [hunk 18]
```diff
@@ -0,0 +1,248 @@
...
+This validates the basic syntactical integrity of the EDI submission.
+
+* Type 2: HIPAA syntactical requirement testing
+
+Testing for HIPAA Implementation Guide-specific syntax requirements, such as limits on repeat counts, used and not used qualifiers, codes, elements, and segments. Also included in this type is testing for HIPAA required or intra-segment situational data elements, testing for non-medical code sets as laid out in the WEDI SNIP implementation guide, and values and codes noted in the WEDI SNIP implementation guide using an X12 code list or table.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-hipaa.adoc:238]:
```suggestion
Tests for HIPAA Implementation Guide-specific syntax requirements, such as limits on repeat counts, used and not used qualifiers, codes, elements, and segments. Also included in this type is testing for HIPAA-required or intra-segment situational data elements, testing for non-medical code sets as laid out in the WEDI SNIP implementation guide, and values and codes noted in the WEDI SNIP implementation guide using an X12 code list or table.
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 19]
```diff
@@ -0,0 +1,241 @@
...
+
+== Default Configuration
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+|Name | String | The name for this configuration. Connectors reference the configuration with this name. | |x
+| Expiration Policy a| ExpirationPolicy |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. This does not mean that the platform expires the instance at the exact moment that it becomes eligible. The runtime actually purges the instances as it sees fit. |  |
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-reference.adoc:12]:
```suggestion
| Expiration Policy a| ExpirationPolicy |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. |  |
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-reference.adoc [hunk 20]
```diff
@@ -0,0 +1,241 @@
...
+** PLATFORM
+** ASCII
+** ISO8859_1
+** IBM1047
+** UTF8 | Character encoding for messages. This character encoding is always used for both send and receive messages. |  PLATFORM |
+| Character set a| Enumeration, one of:
+
+** BASIC
+** EXTENDED
+** UNRESTRICTED | Characters allowed in string data. Invalid characters are replaced by the substitution character, if used, or rejected as errors if no substitution character is set or enabled for receive messages in parser options. Either way the invalid characters are logged, and for receive messages are reported in 997 functional acknowledgments. |  EXTENDED |
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-reference.adoc:36]:
```suggestion
** UNRESTRICTED | Characters allowed in string data. Invalid characters are replaced by the substitution character, if used, or rejected as errors if no substitution character is set or enabled for receive messages in parser options. Either way the invalid characters are logged, and receive messages are reported in 997 functional acknowledgments. |  EXTENDED |
```

Comment (valkyrie69) [reply x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-reference.adoc:36]: ?

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-understand-edi-schema.adoc [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+
+* segments, composites, and elements
+
+ESLs for the all xref:x12-edi-versions-x12.adoc[supported X12 versions] are included.
+If you are using HIPAA, then a set of xref:x12-edi-versions-hipaa.adoc[HIPAA variations] of base X12 transaction sets are included.
+
+You can customize X12 schema definitions to suit your data by defining an overlay schema. An overlay schema is a type of ESL that enables you to modify a base schema, such as an X12 4010 850 schema, with your specific conventions.
+You don't need an overlay schema if you're using the structure defined by the standard, but most EDI exchanges modify the base definitions in an implementation convention.
+
+You can define your own schemas from scratch, or copy and edit a base X12 schema definition to suit your data as shown in the xref:x12-edi-schema-language-reference.adoc[EDI Schema Language Reference].
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-connector-understand-edi-schema.adoc]:
```suggestion
You can define your own custom schemas, or copy and edit a base X12 schema definition to suit your data as shown in the xref:x12-edi-schema-language-reference.adoc[EDI Schema Language Reference].
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-schema-language-reference.adoc [hunk 22]
```diff
@@ -0,0 +1,385 @@
...
+* Anypoint Connector for X12 EDI (X12 EDI Connector)
+* Anypoint Connector for EDIFACT (EDIFACT Connector)
+* Anypoint Connector for TRADACOMS (TRADACOMS Connector)
+* Anypoint Connector for HL7 (HL7 Connector)
+
+== Components of an ESL Document
+
+The top-level definitions in an ESL document show:
+
+* the EDI standard
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-schema-language-reference.adoc:17]: I think all bullet points should be init capped.

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-schema-language-reference.adoc [hunk 23]
```diff
@@ -0,0 +1,385 @@
...
+
+=== Readability
+
+YAML uses a combination of lists and sets of key-value pairs. The order of
+values is not important, as long as the required items are present. Quotes
+(such as single or double) are used around values that consist
+of digits but are meant to be interpreted as strings (otherwise the
+YAML parser treats the values as numbers). Indentation is used to show the nesting of lists.
+
+The ESL structures shown here defines all simple key-value pairs before any lists that are part of the same definition.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-schema-language-reference.adoc]:
```suggestion
The ESL structures shown here define all simple key-value pairs before any lists that are part of the same definition.
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-schema-language-reference.adoc [hunk 24]
```diff
@@ -0,0 +1,385 @@
...
+|Structure Key/Section |Description
+|id |Structure identifier
+|name |Structure name
+|class |A class of structure that is equivalent to an X12 functional group identifier. This is ignored in EDIFACT structure definitions.
+|heading |List of segments and groups within the heading section of the structure.
+|detail |List of segments and groups within the detail section of the structure.
+|summary |List of segments and groups within the heading section of the structure.
+|===
+
+The lists of segments for each section of the structure use the same form. Each item in the list is either a segment reference or a group definition. Segment references use a compact YAML syntax: the values for each are shown as comma-separated key-value pairs enclosed in curly braces.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-schema-language-reference.adoc:95]:
```suggestion
The lists of segments for each section of the structure use the same form. Each item in the list is either a segment reference or a group definition. Segment references use a compact YAML syntax; the values for each are shown as comma-separated key-value pairs enclosed in curly braces.
```

File: x12-edi/2.6/modules/ROOT/pages/x12-edi-schema-language-reference.adoc [hunk 25]
```diff
@@ -0,0 +1,385 @@
...
+
+[%header, cols="20a,80a"]
+|===
+|Name |Description
+|id |composite identifier
+|name |composite name
+|values |list of elements and composites within the composite
+|===
+
+The values list references elements and composites by ID, and use a compact YAML syntax: the values for each are shown as comma-separated key-value pairs enclosed in curly braces.
```
Comment (valkyrie69) [inline x12-edi/2.6/modules/ROOT/pages/x12-edi-schema-language-reference.adoc:208]: I'm not sure about using a colon that way. I commented previously, but disregard if you think it's fine. And it's really not important right now.

**Review by valkyrie69 — DISMISSED** [review]

**Review by valkyrie69 — DISMISSED** [review]

**Review by valkyrie69 — APPROVED** [review]

**Pull request merged**